File: model/src/main/scala/ru/mipt/acsl/decode/model/component/message/MessageParameterRef.scala
package ru.mipt.acsl.decode.model.component.message

import ru.mipt.acsl.decode.model.component.Component
import ru.mipt.acsl.decode.model.types.{DecodeType, StructField}


trait MessageParameterRef {
  def component: Component

  def structField: Option[StructField]

  def path: MessageParameterPath

  def resultType: DecodeType

  def t: DecodeType
}

File: model/src/main/java/ru/mipt/acsl/decode/model/component/message/MessageParameterRef.java
package ru.mipt.acsl.decode.model.component.message;

import ru.mipt.acsl.decode.model.component.Component;
import ru.mipt.acsl.decode.model.component.MessageParameterPath;
import ru.mipt.acsl.decode.model.types.DecodeType;
import ru.mipt.acsl.decode.model.types.StructField;

import java.util.Optional;

/**
 * @author Artem Shein
 */
public interface MessageParameterRef {

    Component component();

    Optional<StructField> structField();

    MessageParameterPath path();

    DecodeType resultType();

    DecodeType t();

}
